### ISSUE TITLE
`[<AttachMembers>]` not compatible with f# `member this.Item`

### ISSUE THREAD

**Issue by Freymaurer** [fable-compiler/Fable#3571]
### Description

Use `[<AttachMembers>]` to improve js syntax, `member this.Item ...` to make `classInstance.[0]` syntax accessible.

### Repro code

```fsharp
module Class

open Fable.Core
open System.Collections.Generic

[<AttachMembers>]
type MyType() =
    member val MyNumbers = ResizeArray [0..100] with get, set

    member this.Item 
        with get(index) = 
            this.MyNumbers.[index] 

let x = MyType()

x.[0] |> printfn "%i"
// Uncaught TypeError: x.get_Item is not a function
```

Link to the [REPL](https://fable.io/repl/#?code=LYewJgrgNgpgBAYSgQwM6oLAChsgA4wB2cAYsgEawB0CIATjLgcQMoCeqALjMDSFLADGnAJYhCqKgHEiMOiMHZsAbQA8AQU6dkggBYBZHuTmoAfAF1snNgTj62AFRswAFAEo4AXmxxfcYEZycABuyFB2bAByEMDGdKhecABKMKgiAF4w6nR0yGxwygAMVFQAjIWF5nAA7iKcunAA5jCcADRwqC1KWH7+gXRw9SKSAJLcwHA+vX619U0tLiKEYDAAHh6ekz3TO0OS9tGxJlTKSyurVd2wnHCrifZOBO7dqyeVcAA+pnB48oScADNiAAiACkImBQA&html=Q&css=Q).

### Expected and actual results

Returned js code implements `Item` member, but calls `x.get_Item(0)` for `x.[0]`, instead `x.Item(0)`

```js
    Item(index) {
        const this$ = this;
        return this$.MyNumbers[index] | 0;
    }
```

The cleanest solution in my opinion would be to enable js array access syntax for classes implementing the Item member. But i am not sure if this can be done.

```js
x[0]
```

**Comment by ncave** [comment]
@Freymaurer Just tested it with latest Fable (4.4.1), seems to work fine:

```js
(function () {
    const arg = x.Item(0) | 0;
    toConsole(printf("%i"))(arg);
})();
```
I guess the REPL is running an older Fable build, so it will resolve itself once it's refreshed.
Feel free to reopen if you still have issues.

**Comment by Freymaurer** [comment]
> @Freymaurer Just tested it with latest Fable (4.4.1), seems to work fine:

Ah yes! after updating fable my tests pass! Thanks!

> Where the call to the getter will actually call the setter, without errors (cause JavaScript) but it's incorrect.
I'll keep it open until it's fixed.

Good to know! Thanks for looking into this!

**Comment by Freymaurer** [comment]
> Where the call to the getter will actually call the setter, without errors (cause JavaScript) but it's incorrect.
I'll keep it open until it's fixed.

I just ran into this exact issue again 😅 Can't post a REPL as it is still v4.1.*, but here is the code i used for testing:

```fsharp
#r "nuget: Fable.Core, 4.2.0"

open Fable.Core

[<AttachMembers>]
type TestClass(?arg: seq<string>) =
    let arg = 
        arg 
        |> Option.map ResizeArray 
        |> fun opt -> defaultArg opt (ResizeArray())
    member val StringCounter = arg with get, set
    member this.Item
        with get(index:int) =
            printfn "GET" 
            this.StringCounter.[index]
        and set(index:int) (v:string) = 
            printfn "SET" 
            this.StringCounter.[index] <- v

let testInstance = TestClass([for i in 0..5 do yield $"{i}"])

testInstance[5]
testInstance[3] <- "Hello World"
```

Both getter and setter transpile to `Item` with different args. Therefore the setter still shadows the getter and only the setter is called:

![image](https://github.com/fable-compiler/Fable/assets/39732517/008b3010-7999-48b0-9fea-53c709420943)

**Comment by ncave** [comment]
@Freymaurer It's already fixed by #3573, will be in the next release.

**Comment by Freymaurer** [comment]
So in 4.4.2? Do you know when this will be released?

**Comment by ncave** [comment]
@Freymaurer 
> So in 4.4.2? Do you know when this will be released?

I don't think we have a fixed cadence for releases, so I'm assuming it will be when we accumulate enough changes to do a release.

++ @MangelMaxime can correct me if I'm wrong.

**Comment by MangelMaxime** [comment]
Hello @Freymaurer @ncave ,

indeed, we don't have a fixed cadence for the release. In general, we wait to have enough changes or a fix for a nasty bug before releasing. 

We could release a new version each time we make a changes but I feel like it would be less impactful and would also generate a lot of noise in term of tweets etc.

With that said, I released a new version of Fable (4.5.0) which should fix the problem reported in this issue.
